Document Convergent Tokenization and Token Lookup (#15819)
hashicorp/vault: website/content/api-docs/secret/transform.mdx

}
```

## Lookup Token

This endpoint returns the token given a plaintext and optionally an
expiration or range of expirations.  This operation is only supported

> of the transform is configured as 'convergent', or if the mapping

Reviewer: ```suggestion
if the transformation is configured as 'convergent', or if the mapping
```

mode is exportable and the storage backend is external.  Tokens may be
looked up with an explicit expiration, any expiration, or with range
of acceptable expiration times. This endpoint is only valid for tokenization 
transformations.

| Method | Path                              |
| :----- | :-------------------------------- |
| `POST` | `/transform/tokens/:role_name` |

### Parameters

- `role_name` `(string: <required>)` –
  Specifies the role name to use for this operation. This is specified as part
  of the URL.

- `value` `(string: <required>)` –

>   Specifies the token to test for whether it has a valid tokenization.

Reviewer: ```suggestion
  Specifies the plaintext of tokens to search for.
```

- `expiration` `(string: "")` - The precise expiration of the token.  If omitted, 
  this specifically searches for tokens with no expiration.  If the string
  "any", will return tokens with any or no expiration.  Otherwise,
  the string must be the RFC3339 formatted time and date of expiration. `expiration`
  may not be used at the same time as `min_expiration` and `max_expiration`.

- `min_expiration` `(string: "")` - The minimum expiration time of the token,
  inclusive, as an RFC3339 formatted time and date. 

>   `min_expiration` may not be used at the same time as `expiration`.

Reviewer: ```suggestion
  `min_expiration` may not be used at the same time as `expiration`.
  When provided, `max_expiration` must also be provided.
```

- `max_expiration` `(string: "")` - The maximum expiration time of the token,
  inclusive, as an RFC3339 formatted time and date. 

>   `max_expiration` may not be used at the same time as `expiration`.

Reviewer: ```suggestion
  `max_expiration` may not be used at the same time as `expiration`.
  When provided, `min_expiration` must also be provided.
```

- `transformation` `(string)` –
  Specifies the transformation within the role that should be used for this

>   decode operation. If a single transformation exists for role, this parameter

Reviewer: ```suggestion
  lookup operation. If a single transformation exists for role, this parameter
```

  may be skipped and will be inferred. If multiple transformations exist, one
  must be specified.

- `reference` `(string: "")` -
  A user-supplied string that will be present in the `reference` field on the
  corresponding `batch_results` item in the response, to assist in understanding
  which result corresponds to a particular input. Only valid on batch requests
  when using ‘batch_input’ below.

- `batch_input` `(array<object>: nil)` -
  Specifies a list of items to be decoded in a single batch. When this

>   parameter is set, the 'value', 'transformation', and 'reference' parameters are

Reviewer: ```suggestion
  parameter is set, the `value`, `transformation`, and `reference` parameters are
```

}
```

## Lookup Token

> This endpoint returns the token given a plaintext and optionally an

Reviewer 2: I'm not entirely sure I understand the first sentence.

> expiration or range of expirations.  This operation is only supported

Reviewer 2: 'This' as in 'this operation' is an antecedent so I'm not sure what 'this' is is referring to. If it's lookup token, perhaps it might be better to say Lookup Token is only supported...
